File: Easy/JavaScript/2078. Two Furthest Houses With Different Colors.js
var maxDistance = function(colors) {
    let n = colors.length;
    let i = 0;
    let j = n - 1;

    while(colors[0] == colors[j])
        j--;
    while(colors[n - 1] == colors[i])
        i++;
    
    return Math.max(j, n - i - 1);
};

File: Easy/Python/2078. Two Furthest Houses With Different Colors.py
class Solution(object):
    def maxDistance(self, colors):
        n = len(colors)
        i = 0
        j = n - 1

        while colors[0] == colors[j]:
            j -= 1
        while colors[n - 1] == colors[i]:
            i += 1
        
        return max(j, n - i - 1)
